src/org/infoglue/calendar/actions/CreateResourceAction.java

```diff
@@ -182,9 +163,16 @@ else if(name.equals("mode"))
 		}
         
         return Action.SUCCESS;
-    } 
-    
-    public String getEventIdAsString()
+    }
+
+	private boolean validateFileType(DiskFileItem dfi) {
+		String fileType = dfi.getContentType();
+		log.info("Uploaded file's file type is: " + fileType);
+		List<String> allowedFileType = ResourceController.getController().getFileTypesForAssetKey(getAssetKey());
+		return allowedFileType.contains(fileType);
+	}
```

Comment:
It's very easy to manipulate the HTTP request before sending it. You can read more here for example: http://resources.infosecinstitute.com/file-upload-vulnerabilities/

If this code only should prevent mistakes, I think it's good enough. To provide security, it depends on how the file is going to be used, where it will be placed in the file system, and so on.